.github/workflows/utils.yml

```diff
@@ -0,0 +1,235 @@
+name: utils
+
+on:
+  workflow_call:
+
+jobs:
+  test_valgrind:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - name: Install dependencies
+        id: install_dependencies
+        run: sudo apt-get install -y lcov cmake gcc g++ libclang-dev valgrind
+
+      - name: Run tests
+        run: |
+          mkdir build
+          cd build
+          cmake ../ -DBUILD_PLUGIN_NOOP=ON -DENABLE_TESTS=ON -DBUILD_EXAMPLES=ON -DCMAKE_CXX_COMPILER=g++-12 
+          make && make test
+
+      - name: Run valgrind tests (core)
+        run: |
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_id_pool
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_log
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_misc_mock
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_misc
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_plugin
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_resource
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_session
+          valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes  ./build/test/test_vaccel
+
+      - name: Run valgrind tests (ops)
+        run: |
+          cd build/test/  
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_blas
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_exec
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_image
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_image_generic  
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_image   
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_minmax     
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_noop        
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_tf_interference        
+          env VACCEL_BACKENDS=../plugins/noop/libvaccel-noop.so valgrind --leak-check=full --show-leak-kinds=all --track-origins=yes ./test_tf_model
```

Comment:
we could address these linting comments (the trailing whitespace in the utils.yml file maybe)